web/src/components/apps/AppVersionHistoryRow.jsx

```diff
@@ -434,15 +436,22 @@ class AppVersionHistoryRow extends Component {
                 "primary blue": isPrimaryButton,
               })}
               disabled={this.isActionButtonDisabled(version)}
-              onClick={() =>
-                needsConfiguration
-                  ? this.props.history.push(
-                      `/app/${app.slug}/config/${version.sequence}`
-                    )
-                  : isRollback
-                  ? actionFn(version, true)
-                  : actionFn(version)
-              }
+              onClick={() => {
+                this.props.handleActionButtonClicked();
+                if (needsConfiguration) {
```

Comment:
can this be extracted into a function for better readability?

Reply:
it should be i'm going to refactor this component